### PR TITLE
Make SSESessionManager ctor free threaded

### PR DESCRIPTION
**Pull request by bigfluffycookie**
part of https://github.com/SonarSource/sonarlint-visualstudio/issues/4856

### CODE REVIEW

File: src/ConnectedMode/ServerSentEvents/SSESessionManager.cs [hunk 1]
```diff
@@ -27,16 +27,9 @@
...
 {
-    /// <summary>
-    /// Reacts to project changes and opens/closes Server Sent Events sessions
-    /// </summary>
-    internal interface ISSESessionManager : IDisposable
-    {
-    }
-
-    [Export(typeof(ISSESessionManager))]
+    [Export(typeof(SSESessionManager))]
```
Comment (bigfluffycookie) [inline src/ConnectedMode/ServerSentEvents/SSESessionManager.cs:30]: We dont use this anywhere else so I removed the interface so it is the same as the TriggerImportInstall class

File: src/ConnectedMode/ConnectedModePackage.cs [hunk 2]
```diff
@@ -84,17 +84,24 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
...
-            // See https://github.com/SonarSource/sonarlint-visualstudio/issues/3886
+            logger.WriteLine(Resources.Package_Initialized);
+        }
+
+        /// <summary>
+        /// Trigger an initial update of suppressions (These classes might have missed the initial solution binding
+        /// event from the ActiveSolutionBoundTracker)
+        /// See https://github.com/SonarSource/sonarlint-visualstudio/issues/3886
+        /// </summary>
+        private void LoadServicesAndDoInitialUpdates(IComponentModel componentModel)
```
Comment (bigfluffycookie) [inline src/ConnectedMode/ConnectedModePackage.cs:95]: With how many there are now I thought it made sense to put it in a seperate method.

File: src/ConnectedMode/ConnectedModePackage.cs [hunk 3]
```diff
@@ -64,7 +64,7 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
 
             logger.WriteLine(Resources.Package_Initializing);
 
-            sseSessionManager = componentModel.GetService<ISSESessionManager>();
+            LoadServicesAndDoInitialUpdates(componentModel);
```
Comment (bigfluffycookie) [inline src/ConnectedMode/ConnectedModePackage.cs:67]: I moved it up as on line 85 there is an await and this way it doesnt disturb the load order as much

**Comment by bigfluffycookie** [comment]
A little confused about the dispose code smell as I didnt change anything on that, should I fix it?

File: src/ConnectedMode/ServerSentEvents/SSESessionManager.cs [hunk 4]
```diff
@@ -27,16 +27,9 @@
 
 namespace SonarLint.VisualStudio.ConnectedMode.ServerSentEvents
 {
-    /// <summary>
-    /// Reacts to project changes and opens/closes Server Sent Events sessions
-    /// </summary>
-    internal interface ISSESessionManager : IDisposable
```
Comment (duncanp-sonar) [inline src/ConnectedMode/ServerSentEvents/SSESessionManager.cs:33]: I think you need to make `SSESessionManager : IDisposable` now (one of the code smells).

File: src/ConnectedMode/ConnectedModePackage.cs (hunk 2 above)
Comment (duncanp-sonar) [reply src/ConnectedMode/ConnectedModePackage.cs:95]: The doc comment on the method needs to updated.

**Review by duncanp-sonar — COMMENTED** [review]
Looks good. A couple of minor comments.

**Review by duncanp-sonar — APPROVED** [review]

**Pull request merged**